File: VB/VB/WinWebSolution.Module/MyNonPersistentDomainObject.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl

Namespace WinWebSolution.Module
	<DefaultClassOptions, NonPersistent> _
	Public Class MyNonPersistentDomainObject
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Public Sub New(ByVal session As Session, ByVal name As String)
			MyBase.New(session)
			Me.Name = name
		End Sub
		Private _name As String
		Public Property Name() As String
			Get
				Return _name
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", _name, value)
			End Set
		End Property
	End Class
End Namespace

File: CS/8.3/WinWebSolution.Module/MyNonPersistentDomainObject.cs
using DevExpress.Xpo;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;

namespace WinWebSolution.Module {
    [DefaultClassOptions]
    [NonPersistent]
    public class MyNonPersistentDomainObject : BaseObject {
        public MyNonPersistentDomainObject(Session session) : base(session) { }
        public MyNonPersistentDomainObject(Session session, string name):base(session) {
            this.Name = name;
        }
        private string _name;
        public string Name {
            get { return _name; }
            set { SetPropertyValue("Name", ref _name, value); }
        }
    }
}
